### PR TITLE
Remove rewards staking process

### PR DESCRIPTION
**Pull request by foxpy**
Progress so far:
- [x] set withdrawal_address when granting authz
- [x] pump setup for staking rewards
- [x] remove staking_rewards state
- [ ] redo non-native rewards handling
- [x] reroute rewards through splitter and staker
- [x] fix unit tests
- [x] fix integration tests

### CODE REVIEW

File: contracts/puppeteer/src/contract.rs [hunk 1]
```diff
@@ -195,11 +196,7 @@ pub fn execute(
 ) -> ContractResult<Response<NeutronMsg>> {
     let puppeteer_base = Puppeteer::default();
     match msg {
-        ExecuteMsg::Delegate {
-            items,
-            fee,
-            reply_to,
-        } => execute_delegate(deps, info, items, fee, reply_to),
+        ExecuteMsg::Delegate { items, reply_to } => execute_delegate(deps, info, items, reply_to),
```
[the remark this answers is not in the document]
Comment (foxpy) [reply contracts/puppeteer/src/contract.rs:199]: This message is only referenced in tests and in hook-tester contract. Hook-tester itself is newer called in a way to trigger puppeteer's delegate message.  I guess it is safe to delete it now.

File: contracts/splitter/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Token distributor
```
Comment (albertandrejev) [inline contracts/splitter/README.md:1]: Splitter or distributor?

File: contracts/splitter/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    CONFIG.save(deps.storage, &new_config)?;
+    Ok(Response::default())
+}
+
+pub fn execute_distribute(deps: DepsMut, env: Env, _info: MessageInfo) -> ContractResult<Response> {
+    let config: Config = CONFIG.load(deps.storage)?;
+    let total_share = config
+        .receivers
+        .iter()
+        .fold(Uint128::zero(), |acc, (_, share)| acc + share);
```
Comment (albertandrejev) [inline contracts/splitter/src/contract.rs:68]: you can simplify this a little with `map`,`sum`

File: contracts/splitter/src/contract.rs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                to_address: receiver.to_string(),
+                amount: vec![Coin::new((share * k).u128(), config.denom.to_string())],
+            })
+        })
+        .collect::<Vec<_>>();
+    Ok(Response::default().add_messages(messages))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> ContractResult<Response> {
```
Comment (albertandrejev) [inline contracts/splitter/src/contract.rs:94]: can you implement our default migrate function?

File: contracts/factory/src/contract.rs [hunk 5]
```diff
@@ -574,3 +624,20 @@ pub fn migrate(
...
     Ok(Response::new())
 }
+
+fn get_splitter_receivers(
+    fee_params: Option<crate::msg::FeeParams>,
+    staker_address: String,
+) -> ContractResult<Vec<(String, cosmwasm_std::Uint128)>> {
+    match fee_params {
+        Some(fee_params) => {
+            let fee_weight = Uint128::from(10000u128) * fee_params.fee;
```
[the remark this answers is not in the document]
Comment (ratik) [reply contracts/factory/src/contract.rs:634]: this way we can achieve disctribution part like 0.01%
anyway we can update weights after the protocol is deployed

File: contracts/factory/src/contract.rs [hunk 6]
```diff
@@ -300,7 +332,7 @@ pub fn instantiate(
             label: get_contract_label("withdrawal-voucher"),
             msg: to_json_binary(&WithdrawalVoucherInstantiateMsg {
                 name: "Drop Voucher".to_string(),
-                symbol: "LDOV".to_string(),
+                symbol: "DROPV".to_string(),
```
[the remark this answers is not in the document]
Comment (ratik) [reply contracts/factory/src/contract.rs:335]: let's have a task in backlog for this one

**Review by albertandrejev — APPROVED** [review]

**Pull request merged**